homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,181 @@
+"""Support for Alpha2 room control unit via Alpha2 base."""
+import logging
+
+import aiohttp
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import Alpha2BaseCoordinator
+from .const import DOMAIN, PRESET_AUTO, PRESET_DAY, PRESET_NIGHT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add Alpha2Climate entities from a config_entry."""
+
+    coordinator: Alpha2BaseCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        Alpha2Climate(coordinator, heatarea_id) for heatarea_id in coordinator.data
+    )
+
+
+# https://developers.home-assistant.io/docs/core/entity/climate/
+class Alpha2Climate(CoordinatorEntity, ClimateEntity):
+    """Alpha2 ClimateEntity."""
+
+    target_temperature_step = 0.2
+
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_hvac_modes = [HVAC_MODE_HEAT, HVAC_MODE_COOL]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_preset_modes = [PRESET_AUTO, PRESET_DAY, PRESET_NIGHT]
+
+    def __init__(self, coordinator: Alpha2BaseCoordinator, heatarea_id: str):
+        """Initialize Alpha2 ClimateEntity."""
+        super().__init__(coordinator)
+        # Set the right type for mypy
+        self.coordinator: Alpha2BaseCoordinator = self.coordinator
+        self.heatarea_id = heatarea_id
```

Comment:
We prefer to separate words with underscore in variable names.
```suggestion
        self.heat_area_id = heat_area_id
```